### PR TITLE
remove gofr's dependency from ftp package

### LINKED ISSUE #1259: Remove datasources' dependency on GoFr
Few datasources like ftp, sftp, nats import GoFr. This dependency must be removed to ensure that datasources are not dependent on any particular versions of GoFr.

### PR DESCRIPTION
**Pull request by Umang01-hash**
## Pull Request Template

**Description:**

-   Resolves #1259 partially.

**Checklist:**

-   [ ] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [ ] All new code is covered by unit tests.
-   [ ] This PR does not decrease the overall code coverage.
-   [ ] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

File: pkg/gofr/datasource/file/ftp/interface.go [hunk 1]
```diff
@@ -3,11 +3,82 @@ package ftp
...
 	"context"
 	"io"
+	"os"
 	"time"
 
 	"github.com/jlaffaye/ftp"
 )
 
+// FTPFile represents a file in the filesystem.
+type FTPFile interface {
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/interface.go:13]: There is no need to name this FTPFile in ftp package


```suggestion
// File represents a file in the filesystem.
type File interface {
```

Especially when next interface is named FileInfo

File: pkg/gofr/datasource/file/ftp/interface.go [hunk 2]
```diff
@@ -3,11 +3,82 @@ package ftp
...
+	IsDir() bool        // abbreviation for Mode().IsDir()
+}
+
+type RowReader interface {
+	Next() bool
+	Scan(interface{}) error
+}
+
+// FTPFileSystem : Any simulated or real filesystem should implement this interface.
+type FTPFileSystem interface {
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/interface.go:38]:
```suggestion
// FileSystem : Any simulated or real filesystem should implement this interface.
type FileSystem interface {
```

File: pkg/gofr/datasource/file/ftp/file.go [hunk 3]
```diff
@@ -14,8 +14,6 @@ import (
 	"time"
 
 	"github.com/jlaffaye/ftp"
-
-	file_interface "gofr.dev/pkg/gofr/datasource/file"
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/file.go:18]: Is this file still needed then?

File: pkg/gofr/datasource/file/ftp/interface.go (hunk 1 above)
Comment (Umang01-hash) [reply pkg/gofr/datasource/file/ftp/interface.go]: I completely agree with your point but we can't keep it just File as then it is colliding with the struct named `File` that has all the implementation of the interface.

File: pkg/gofr/datasource/file/ftp/interface.go (hunk 2 above)
Comment (Umang01-hash) [reply pkg/gofr/datasource/file/ftp/interface.go]: `FileSystem` is also a struct so we need to rename it to `FTPFileSystem`

File: pkg/gofr/datasource/file/ftp/file.go (hunk 3 above)
Comment (Umang01-hash) [reply pkg/gofr/datasource/file/ftp/file.go:18]: no with the updated implementation that's not needed

**Comment by Umang01-hash** [comment]
The failing `S3` tests and build for the workflow run can be addressed by this PR: https://github.com/gofr-dev/gofr/pull/1324
